app/helpers/content_header_helper.rb

```diff
@@ -0,0 +1,211 @@
+      raw(strip_tags(string)).strip
+    end
+  end
+
+  class ContentToolbar < AbstractContent
+    def button(text, location, options = {})
+      options[:class] = %w(button) + Array(options[:class])
+      item text, location, options
+    end
+
+    def item(text, location, options = {})
+      return '' unless show?(options)
+      link_options = extract_from options
+      toolbar_item do
+        link_to location, link_options do
+          concat button_icon(options)
+          concat button_text(options.merge text: text)
+        end
+      end
+    end
+
+    def toolbar_item(options = {})
+      options[:class] = Array(options[:class]) + %w(toolbar-item)
+      options[:role] = 'menuitem'
+      if block_given?
+        content_tag :li, options do
+          yield
+        end
+      else
+        content_tag :li, '', options
+      end
+    end
+
+    def watch_button(object, user, options = {})
+      watch_text = options.delete(:watch_text) || I18n.t(:button_watch)
+      unwatch_text = options.delete(:unwatch_text) || I18n.t(:button_unwatch)
+      return '' unless object.respond_to?(:watched_by?) || user.anonymous?
+      watched = object.watched_by?(user)
+      text = watched ? unwatch_text : watch_text
+      method = watched ? :delete : :post
+      icon = watched ? :'not-watch' : :'watch-1'
+      additionals = {
+        data: {
+          remote: true,
+          method: method,
+          watch_text: watch_text,
+          unwatch_text: unwatch_text,
+          watch_icon: :'watch-1',
+          unwatch_icon: :'not-watch',
+          watch_path: watch_path(object, false),
+          unwatch_path: watch_path(object, true),
+          watch_method: :post,
+          unwatch_method: :delete
+        },
+        icon: icon
+      }
+      button text, watch_path(object, watched), options.merge(additionals)
+    end
+
+    def submenu(options = {}, &block)
+      toolbar_item class: '-with-submenu' do
+        link_options = extract_from options
+        link_options[:class] += %w(button)
+        link = link_to('#', link_options) do
+          concat button_icon options
+          concat button_text text: options[:title]
+          concat content_tag :i, '', class: 'button--dropdown-indicator'
+        end
+        concat link
+        css = Array(options[:class]) + %w(toolbar-submenu)
+        css += %w(-last) if options.delete :last
+        concat content_tag :ul, capture(&block), class: css
+      end
+    end
+
+    def submenu_item(text, location, options = {})
+      return '' unless show?(options)
+      unless options[:icon]
+        options[:class] = Array(options[:class]) + %w(no-icon)
+      end
+      toolbar_item_options = options.dup.delete_if { |k, _| k == :icon }
+      item = toolbar_item(toolbar_item_options) do
+        link_options = extract_from(options)
+        link_to location, link_options do
+          concat content_tag :i, '', class: "icon icon-#{options[:icon]}" if options[:icon]
```

Comment:
Line is too long. [91/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.